server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+// We use a child_process because we want to be able to exit the process after
+// we receive the SSR results. We don't want Hydrogen to keep running after we
+// get our initial rendered HTML.
+
+const fork = require('child_process').fork;
+
+const RethrownError = require('../lib/rethrown-error');
+
+// The render should be fast. If it's taking more than 5 seconds, something has
+// gone really wrong.
+const RENDER_TIMEOUT = 5000;
+
+async function renderHydrogenToString(options) {
+  try {
+    let data = '';
+    let childErrors = [];
+
+    const controller = new AbortController();
+    const { signal } = controller;
+    // We use a child_process because we want to be able to exit the process after
+    // we receive the SSR results.
+    const child = fork(
+      require.resolve('./2-render-hydrogen-to-string-fork-script'),
+      [JSON.stringify(options)],
```

Comment:
Passing the options via argv doesn't work with big payloads, `Error: spawn E2BIG`

Solved in https://github.com/matrix-org/matrix-public-archive/pull/37